### PR TITLE
fix: add normalization in get_random_weighted_columns

### PR DESCRIPTION
**Pull request by olivierabz**
Adds a normalization step in `get_random_weighted_columns ` (used when the arg `use_max_modalities = False`, which is how it is used in avatar).
Ensures that the max cumsum value is 1 and that all modalities can be sampled.

### CODE REVIEW

File: saiph/inverse_transform_test.py [hunk 1]
```diff
@@ -31,6 +31,13 @@ def test_get_random_weighted_columns(weights: List[float], expected_index: int)
     assert result.values[0] == expected_index
 
 
+def test_normalization() -> None:
+    """Verify that the last modality can be sampled when its cumulated sum is greater than 1."""
+    df = pd.DataFrame(data=[[0.5, 0.5, 0.5]])
+    result = get_random_weighted_columns(df, np.random.default_rng(4))
+    assert result.values[0] == 2
```
Comment (jpetot) [inline saiph/inverse_transform_test.py:38]: I think we should add other tests to be sure of the `get_random_weighted_columns` behaviour

File: saiph/inverse_transform_test.py [hunk 2]
```diff
@@ -31,6 +31,13 @@ def test_get_random_weighted_columns(weights: List[float], expected_index: int)
     assert result.values[0] == expected_index
 
 
+def test_normalization() -> None:
+    """Verify that the last modality can be sampled when its cumulated sum is greater than 1."""
```
Comment (jpetot) [inline saiph/inverse_transform_test.py:35]: I don't really understand this test

**Review by jpetot — CHANGES_REQUESTED** [review]
I don't really understand the test, and have you created a regression test before to implement this fix ?

File: saiph/inverse_transform_test.py (hunk 2 above)
Comment (olivierabz) [reply saiph/inverse_transform_test.py:35]: The df used in this test is a list of probabilities for hypothetical modalities (3 modalities). We see that the sum of the first 2 is 1, so without normalizing this list, there is no way that the 3rd modality can ever be picked.

That's the regression test: fail without the fix, pass with it.

File: saiph/inverse_transform_test.py (hunk 1 above)
Comment (olivierabz) [reply saiph/inverse_transform_test.py:38]: what do you have in mind ?

**Review by jpetot — APPROVED** [review]
I approve, but I realized there isn't any test of the function `get_random_weighted_columns` before yours. 

Are you sure you didn't break anything?

**Comment by olivierabz** [comment]
> get_random_weighted_columns

there is another test: `test_get_random_weighted_columns` which uses 4 sets of parameters. This was not changed and still passes following the fix

**Comment by jpetot** [comment]
> > get_random_weighted_columns
> 
> there is another test: `test_get_random_weighted_columns` which uses 4 sets of parameters. This was not changed and still passes following the fix

My bad, I was looking for a `test_get_random_weighted_columns.py` file

File: saiph/inverse_transform_test.py [hunk 3]
```diff
@@ -31,6 +31,18 @@ def test_get_random_weighted_columns(weights: List[float], expected_index: int)
     assert result.values[0] == expected_index
 
 
+def test_normalization() -> None:
+    """Verify that the last modality can be sampled when its cumulated sum is greater than 1.
+
+    This test will fail if the values are not normalized first because the last modality would
+    have no chance to be drawn.
```
Comment (tcrasset) [inline saiph/inverse_transform_test.py:38]: Clean!

**Pull request merged**